### PR TITLE
Initial setup of integration testing of the vscode plugin

### PR DESCRIPTION
**Pull request by rictic**
This just launches vscode with our extension and verifies that vscode believes that it's installed.

Based on the docs at https://code.visualstudio.com/api/working-with-extensions/testing-extension

Part of https://github.com/runem/lit-analyzer/issues/222

### CODE REVIEW

File: packages/vscode-lit-plugin/src/test/scripts/mocha-driver.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import * as path from "path";
+import Mocha from "mocha";
+import glob from "glob";
+
+/**
+ * Finds all test files and runs them.
```
Comment (justinfagnani) [inline packages/vscode-lit-plugin/src/test/scripts/mocha-driver.ts:6]: Add to the description where it's looking for tests?

File: packages/vscode-lit-plugin/src/test/scripts/mocha-driver.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ * Should resolve if the tests pass, reject if any fail.
+ */
+export async function run(): Promise<void> {
+	const mocha = new Mocha({
+		ui: "tdd",
+		color: true
+	});
+
+	const testsRoot = path.join(__dirname, "..");
+	const files = glob.sync("**/**-test.js", { cwd: testsRoot });
```
Comment (justinfagnani) [inline packages/vscode-lit-plugin/src/test/scripts/mocha-driver.ts:19]: Should the glob be `"**/*-test.js"`?

File: packages/vscode-lit-plugin/src/test/scripts/test-runner.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+async function main() {
+	try {
+		// Passed to `--extensionDevelopmentPath`
+		const extensionDevelopmentPath = path.resolve(__dirname, "../../../");
+		// The path to the extension test runner script
+		const extensionTestsPath = path.resolve(__dirname, "./mocha-driver");
+
+		// Download VS Code, unzip it and run the integration test
+		await runTests({ extensionDevelopmentPath, extensionTestsPath });
```
Comment (justinfagnani) [inline packages/vscode-lit-plugin/src/test/scripts/test-runner.ts:18]: you might be able to use top-level await now?

**Review by justinfagnani — APPROVED** [review]

File: packages/vscode-lit-plugin/src/test/scripts/test-runner.ts (hunk 3 above)
Comment (rictic) [reply packages/vscode-lit-plugin/src/test/scripts/test-runner.ts:18]: Would be mildly cool, but would need to update the tsconfig for the whole project, and it's another thing that could go wrong.

File: packages/vscode-lit-plugin/src/test/scripts/mocha-driver.ts (hunk 1 above)
Comment (rictic) [reply packages/vscode-lit-plugin/src/test/scripts/mocha-driver.ts]: Done

File: packages/vscode-lit-plugin/src/test/scripts/mocha-driver.ts (hunk 2 above)
Comment (rictic) [reply packages/vscode-lit-plugin/src/test/scripts/mocha-driver.ts]: +1

**Review by justinfagnani — APPROVED** [review]

**Pull request merged**